healthz/clusterstatus/clusterstatus.go

```diff
@@ -0,0 +1,104 @@
+package clusterstatus
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gravitational/satellite/healthz/config"
+	"github.com/gravitational/satellite/healthz/utils"
+)
+
+type ClusterStatus struct {
+	K8S struct {
+		Status       bool
+		NodesTotal   uint
+		NodesHealthy uint
+	}
+	Etcd struct {
+		Status bool
+	}
+	Mux sync.Mutex
+}
+
+const ContextKey = "clusterstatus"
+
+func (c *ClusterStatus) String() string {
+	c.Mux.Lock()
+	defer c.Mux.Unlock()
+	return fmt.Sprintf("{\"K8S\": {\"Status\": %v, \"NodesTotal\": %v "+
+		"\"NodesHealthy\": %v}, \"Etcd\": {\"Status\": %v}}", c.K8S.Status,
+		c.K8S.NodesTotal, c.K8S.NodesHealthy, c.Etcd.Status)
+}
+
+func (c *ClusterStatus) updateK8sClusterStatus() {
+	_, err := utils.ExecCommand("kubectl", "cluster-info")
+	c.Mux.Lock()
+	defer c.Mux.Unlock()
+	c.K8S.Status = (err == nil) // No errors -- healthy
+}
+
+func (c *ClusterStatus) updateEtcdStatus() {
+	_, err := utils.ExecCommand("etcdctl", "cluster-health")
+	c.Mux.Lock()
+	defer c.Mux.Unlock()
+	c.Etcd.Status = (err == nil) // No errors -- healthy
+}
+
+func (c *ClusterStatus) updateK8SNodesStatus() {
+	out, err := utils.ExecCommand("kubectl", "get", "nodes", "--no-headers", "-o=wide")
+	c.Mux.Lock()
+	defer c.Mux.Unlock()
+	if err != nil {
+		c.K8S.Status = false
+	}
+	var nodesTotal uint = 0
+	var nodesReady uint = 0
+	outReader := bytes.NewReader(out)
+	scanner := bufio.NewScanner(outReader)
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+		status := fields[1]
+		nodesTotal++
+		if status == "Ready" {
+			nodesReady++
+		}
+	}
+	c.K8S.NodesTotal = nodesTotal
+	c.K8S.NodesHealthy = nodesReady
+}
+
+func (c *ClusterStatus) DoChecks() {
```

Comment:
can you please make sure that checkers are objects compatible to checkers defined in the library, here's an example of ETCD checker:

https://github.com/gravitational/satellite/blob/master/monitoring/checkers.go#L41

Can you also make sure that checkers are defined in `lib/monitoring` package while this package simply uses them, so we can keep them consistent. E.g. K8s checker and Etcd checkers are already defined in `lib/monitoring`. You should add `NodesThreshold` checker